addons/zotero/models.py

```diff
@@ -216,3 +211,89 @@ def clear_settings(self):
         """Clears selected folder and selected library configuration"""
         self.list_id = None
         self.library_id = None
+
+    def v2_serialization(self, kind, id, name, path, parent=None, provider_list_id=None):
+        return {
+            'addon': 'zotero',
+            'kind': kind,
+            'id': id,
+            'name': name,
+            'path': path,
+            'parent_list_id': parent,
+            'provider_list_id': provider_list_id or id
+        }
+
+    def get_folders(self, path=None, folder_id=None, **kwargs):
+        """
+        Returns Zotero folders at any level.
+        Top level are group/personal libraries.  Secondary levels are folders within those libraries.
+        If path (also known as library id) is specified, then folders are returned from that specific library.
+
+        If no path(library) specified, then all libraries are returned.
+        """
+        if self.has_auth:
+            if path:
+                return self.get_sub_folders(path, folder_id)
+            else:
+                return self.get_top_level_folders(**kwargs)
+        else:
+            raise exceptions.InvalidAuthError()
+
+    def get_top_level_folders(self, **kwargs):
+        """
+        Returns serialized group libraries - your personal library along with any group libraries.
+
+        This is the top-tier of "folders" in Zotero.
+
+        You can use kwargs to refine what data is returned -  how to limit the number of group libraries,
+        whether to return the personal library alongside group_libraries, or append the total library count.
+        """
+        try:
+            # These kwargs are passed in from ZoteroViews > library_list
+            limit = kwargs.get('limit', None)
+            start = kwargs.get('start', None)
+            return_count = kwargs.get('return_count', 'false')
+            append_personal = kwargs.get('append_personal', 'true')
+            # Fetch group libraries
+            libraries = self.api._fetch_libraries(limit=limit, start=start)
+        except (zotero_errors.HTTPError, zotero_errors.UserNotAuthorised, zotero_errors.ResourceNotFound) as error:
+            raise HTTPError(error)
+
+        # Serialize libraries
+        serialized = []
+        for library in libraries[:-1]:
+            data = library['data']
+            serialized.append(self.v2_serialization('library', data['id'], data['name'], str(data['id'])))
+
+        if return_count == 'true':
+            # Return total number of libraries as last item in list
+            serialized.append(libraries[-1])
+
+        if append_personal == 'true':
```

Comment:
Also, checking against the string `'true'` is unintuitive. Can you pass a boolean instead and make the view responsible for casting the query argument as an boolean? ~~It should be enough to just check for the existence of the query parameter.~~

```python
append_personal = 'append_personal' in request.args
```

Or you could use our `is_truthy` function

```python
from api.base.utils import is_truthy

append_personal = is_truthy(request.args.get('append_personal', False))
```

~~Which is more robust, but not strictly necessary...we're only dealing with v1 here 😶~~

Edit: Looks like `append_personal` is always being passed as a query parameter here: https://github.com/CenterForOpenScience/osf.io/pull/8230/files#diff-8f07ac612faecd6edc0973273e1ab962R473

So you will need to use `is_truthy`.